edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/CbsNode.cs

```diff
@@ -88,6 +88,7 @@ public async Task<bool> AuthenticateAsync(string id)
                         {
                             if (!credentialsInfo.IsAuthenticated)
                             {
+                                credentialsInfo.ClientCredentials.ModelId = modelId;
```

Comment:
This part is quite awkward. I would much rather put the modelId in the credentials map initially, but I couldn't think of a clean way of how to do it. 
This also makes modelId need the "set" property everywhere, which is also awkward. 
Open to ideas on it.

Reply:
I'm going to have the same problem with passing through the authchain on the CBS link.  I was going to change this so that the cached dictionary stores the token instead of a credentials object, and then construct the credentials here on the fly.

You can do that refactor here as part of this work if you have the bandwidth.  Looks like it'll make you change simpler over all and save me work when I get to this.  Otherwise I can come back and clean it up later this month.

---
In reply to: [463754436](https://github.com/Azure/iotedge/pull/3323#discussion_r463754436) [](ancestors = 463754436)